src/tracing/job.ts

```diff
@@ -160,12 +162,28 @@ export async function traceWorkflowRunJobs({
       if (workflowRunJobs.jobs[i].conclusion === "skipped") {
         core.info(`Job ${job.id} was skipped, not tracing.`);
       } else {
+        const annotationsResponse: WorkflowRunJobAnnotationsResponse =
+          await octokit.rest.checks.listAnnotations({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            check_run_id: job.id,
+          });
+        core.debug(`JobAnnotations Count: ${annotationsResponse.data.length}`);
+
+        // We only care about failure annotations generated by github files,
+        // excluding excluding annotations from other tools like rspec failures
+        const annotations = annotationsResponse.data.filter(
+          (a: JobAnnotation) =>
+            a.annotation_level === "failure" && a.path === ".github",
+        );
```

Comment:
Based on investigation of some historical jobs, GitHub and Infra-related failures originate from a `path` of `.github`, while an `rspec` failure originates from a `path` of the ruby spec file.

Reply:
1. For the infra-related failures, is it possible to perform additional attribution e.g. which failures originated from GitHub vs. our own infrastructure?
1. Based on the commentary (in docstrings) above, does this mean that we have an infrastructure failure when `github.conclusion == failure` and `github.job.annotations is not null`?

Reply:
>For the infra-related failures, is it possible to perform additional attribution e.g. which failures originated from GitHub vs. our own infrastructure?

This is the hope! However, the messages in these annotations vary and I think we'll need to gather some data and work to classify distinct messages into buckets, eg. the PR description example, `Could not assume role with OIDC...` I think we'd classify as a `infra`-type failure. 

>Based on the commentary (in docstrings) above, does this mean that we have an infrastructure failure when github.conclusion == failure and github.job.annotations is not null?

Following on my clarification above, that's not necessarily the case as we'd need to inspect the annotation message to really determine root cause; the comments/logic here is just to exclude capturing _other_ types of Job failures, like a `rspec` failure — does that make sense?

Reply:
@ekala Following up here, I was able to collect a sample of annotation messages from our recent GHA jobs [here](https://www.notion.so/wrapbook/Add-annotations-to-CI-traces-126fdd08fcdc80cba708d82d9d8a9a66?pvs=4#131fdd08fcdc80ae9ac4ce6381d05af1) — I think we'll be able to attribute some of these as specifically `infra` or `github`-type errors